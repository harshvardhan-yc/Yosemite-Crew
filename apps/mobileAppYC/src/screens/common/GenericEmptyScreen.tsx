import React from 'react';
import {View, Text, Image, StyleSheet, ImageSourcePropType} from 'react-native';
import {SafeArea} from '@/components/common';
import {Header} from '@/components/common/Header/Header';
import {useTheme} from '@/hooks';

interface GenericEmptyScreenProps {
  headerTitle: string;
  emptyImage: ImageSourcePropType;
  title: string;
  subtitle: string;
  showBackButton?: boolean;
}

export const GenericEmptyScreen: React.FC<GenericEmptyScreenProps> = ({
  headerTitle,
  emptyImage,
  title,
  subtitle,
  showBackButton = false,
}) => {
  const {theme} = useTheme();
  const styles = React.useMemo(() => createStyles(theme), [theme]);

  return (
    <SafeArea>
      <Header title={headerTitle} showBackButton={showBackButton} />
      <View style={styles.container}>
        <View style={styles.contentContainer}>
          <Image source={emptyImage} style={styles.emptyImage} />
          <Text style={styles.title}>{title}</Text>
          <Text style={styles.subtitle}>{subtitle}</Text>
        </View>
      </View>
    </SafeArea>
  );
};

const createStyles = (theme: any) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.background,
    },
    contentContainer: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      paddingHorizontal: theme.spacing[6],
    },
    emptyImage: {
      width: 200,
      height: 200,
      resizeMode: 'contain',
      marginBottom: theme.spacing[6],
    },
    title: {
      ...theme.typography.headlineMedium,
      color: theme.colors.secondary,
      marginBottom: theme.spacing[3],
      textAlign: 'center',
    },
    subtitle: {
      ...theme.typography.bodyMedium,
      color: theme.colors.textSecondary,
      textAlign: 'center',
      lineHeight: 22,
    },
  });
