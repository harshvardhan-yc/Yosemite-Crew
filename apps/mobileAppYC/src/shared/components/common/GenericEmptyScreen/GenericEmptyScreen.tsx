import React from 'react';
import {SafeAreaView} from 'react-native-safe-area-context';
import {ScrollView, StyleSheet, Text, View} from 'react-native';
import {useTheme} from '@/hooks';

interface GenericEmptyScreenProps {
  title: string;
  subtitle: string;
}

export const GenericEmptyScreen: React.FC<GenericEmptyScreenProps> = ({
  title,
  subtitle,
}) => {
  const {theme} = useTheme();
  const styles = React.useMemo(() => createStyles(theme), [theme]);

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView
        contentContainerStyle={styles.content}
        showsVerticalScrollIndicator={false}>
        <View style={styles.card}>
          <Text style={styles.title}>{title}</Text>
          <Text style={styles.subtitle}>{subtitle}</Text>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const createStyles = (theme: any) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.background,
    },
    content: {
      flexGrow: 1,
      padding: 24,
      gap: 16,
    },
    card: {
      backgroundColor: theme.colors.cardBackground,
      borderRadius: 20,
      borderWidth: 1,
      borderColor: theme.colors.border,
      padding: 24,
      ...theme.shadows.xs,
    },
    title: {
      ...theme.typography.h3,
      color: theme.colors.secondary,
      marginBottom: 8,
      textAlign: 'left',
    },
    subtitle: {
      ...theme.typography.paragraph,
      color: theme.colors.textSecondary,
    },
  });
