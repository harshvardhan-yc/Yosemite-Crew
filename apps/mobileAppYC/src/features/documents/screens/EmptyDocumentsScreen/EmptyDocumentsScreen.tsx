import React from 'react';
import {GenericEmptyScreen} from '@/shared/screens/common/GenericEmptyScreen';
import {Images} from '@/assets/images';

export const EmptyDocumentsScreen: React.FC = () => {
  return (
    <GenericEmptyScreen
      headerTitle="Documents"
      emptyImage={Images.emptyDocuments}
      title="Meow-nothing here."
      subtitle={`Hey there! Feel free to drop your vaccine or\nmedical reports right here!`}
      showBackButton={false}
    />
  );
};
