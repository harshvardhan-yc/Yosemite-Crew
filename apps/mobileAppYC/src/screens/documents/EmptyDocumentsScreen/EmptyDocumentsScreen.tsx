import React from 'react';
import {View, Text, Image, StyleSheet} from 'react-native';
import {SafeArea} from '@/components/common';
import {Header} from '@/components/common/Header/Header';
import {useTheme} from '@/hooks';
import {Images} from '@/assets/images';

export const EmptyDocumentsScreen: React.FC = () => {
  const {theme} = useTheme();
  const styles = React.useMemo(() => createStyles(theme), [theme]);

  return (
    <SafeArea>
      <Header title="Documents" showBackButton={false} />
      <View style={styles.container}>
        <View style={styles.contentContainer}>
          <Image source={Images.emptyDocuments} style={styles.emptyImage} />
          <Text style={styles.title}>Meow-nothing here.</Text>
          <Text style={styles.subtitle}>
            Hey there! Feel free to drop your vaccine or{'\n'}medical reports right here!
          </Text>
        </View>
      </View>
    </SafeArea>
  );
};

const createStyles = (theme: any) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.background,
    },
    contentContainer: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      paddingHorizontal: theme.spacing[6],
    },
    emptyImage: {
      width: 200,
      height: 200,
      resizeMode: 'contain',
      marginBottom: theme.spacing[6],
    },
    title: {
      ...theme.typography.headlineMedium,
      color: theme.colors.secondary,
      marginBottom: theme.spacing[3],
      textAlign: 'center',
    },
    subtitle: {
      ...theme.typography.bodyMedium,
      color: theme.colors.textSecondary,
      textAlign: 'center',
      lineHeight: 22,
    },
  });
