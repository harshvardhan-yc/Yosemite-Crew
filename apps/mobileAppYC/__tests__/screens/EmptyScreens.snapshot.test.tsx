// __tests__/screens/EmptyScreens.snapshot.test.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import {GenericEmptyScreen} from '@/shared/components/common/GenericEmptyScreen/GenericEmptyScreen';
import {EmptyDocumentsScreen} from '@/features/documents/screens/EmptyDocumentsScreen/EmptyDocumentsScreen';

// Mock navigation
const mockNavigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => mockNavigation,
}));

describe('Empty Screen Snapshots', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('GenericEmptyScreen', () => {
    it('should render with minimal props', () => {
      const tree = renderer.create(
        <GenericEmptyScreen
          title="Empty"
          subtitle="No items found"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render with all props', () => {
      const tree = renderer.create(
        <GenericEmptyScreen
          title="No Documents"
          subtitle="Start by adding your first document"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render with different icon', () => {
      const tree = renderer.create(
        <GenericEmptyScreen
          title="No Companions"
          subtitle="Add a companion to get started"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('EmptyDocumentsScreen', () => {
    it('should render correctly', () => {
      const tree = renderer.create(
        <EmptyDocumentsScreen />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
