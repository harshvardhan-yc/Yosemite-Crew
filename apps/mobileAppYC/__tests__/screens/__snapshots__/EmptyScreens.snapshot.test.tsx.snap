// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Screen Snapshots EmptyDocumentsScreen should render correctly 1`] = `null`;

exports[`Empty Screen Snapshots GenericEmptyScreen should render with all props 1`] = `null`;

exports[`Empty Screen Snapshots GenericEmptyScreen should render with different icon 1`] = `null`;

exports[`Empty Screen Snapshots GenericEmptyScreen should render with minimal props 1`] = `null`;
